pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -640,8 +653,11 @@ func (c *tomlConfig) loadConfig(path string, forceV2 bool) error {
 		registryMap[c.Registries[i].Prefix] = c.Registries[i]
 	}
 
+	prevAliases := c.Aliases // store the aliases so they're not overridden
```

Comment:
This quick hack has become too much. Please separate
- Load+TOML decode+format conversion into a _newly allocated `tomlConfig`→`V2RegistriesConf`.
- Merging of two `V2RegistriesConf` values into the effective config (vaguely `V2RegistriesConf.overrideByConfig(V2RegistriesConf)` or something like that)

(In a separate commit, or maybe even a separate PR, please.)

Reply:
Added a TODO.

Reply:
This is actually a blocker for me. I feel strongly enough about this that I might even help :)

Related concerns:
- Separating internal caches from file format structs
- `TryUpdatingCache` should return the full configuration currently in force — definitely including the `*.d` files, undefined about the new alias.conf file (I vote “let’s not add this, and consider the idea of returning the full configuration deprecated”).